solidity/contracts/PhasedEscrow.sol

```diff
@@ -66,8 +66,96 @@ contract CurveRewardsEscrowBeneficiary is Ownable {
         curveRewards = _curveRewards;
     }
 
-    function __escrowSentTokens(uint256 amount) external {
+    function __escrowSentTokens(uint256 amount) external onlyOwner {
         token.approve(address(curveRewards), amount);
         curveRewards.notifyRewardAmount(amount);
     }
 }
+
+/// @dev Interface of recipient contract for approveAndCall pattern.
+interface IStakerRewards {
+    function receiveApproval(
+        address _from,
+        uint256 _value,
+        address _token,
+        bytes calldata _extraData
+    ) external;
+}
+
+/// @title StakerRewardsBeneficiary
+/// @notice An abstract beneficiary contract that can receive a withdrawal phase
+///         from a PhasedEscrow contract. The received tokens are immediately 
+///         funded for a designated rewards escrow beneficiary contract.
+contract StakerRewardsBeneficiary is Ownable {
+    IERC20 public token;
+    IStakerRewards public stakerRewards;
+
+    constructor(IERC20 _token, IStakerRewards _stakerRewards) public {
+        token = _token;
+        stakerRewards = _stakerRewards;
+    }
+
+    function __escrowSentTokens(uint256 amount) external onlyOwner {
+        bool success = token.approve(address(stakerRewards), amount);
+        require(success, "Token transfer approval failed");
+        
+        stakerRewards.receiveApproval(
+            address(this),
+            amount,
+            address(token),
+            ""
+        );
+    }
+}
+
+/// @title BeaconBackportRewardsEscrowBeneficiary
+/// @notice Transfer the received tokens to a designated
+///         BeaconBackportRewardsEscrowBeneficiary contract.
+contract BeaconBackportRewardsEscrowBeneficiary is StakerRewardsBeneficiary {
+    constructor(IERC20 _token, IStakerRewards _stakerRewards)
+        public
+        StakerRewardsBeneficiary(_token, _stakerRewards)
+    {
+        token = _token;
+        stakerRewards = _stakerRewards;
+    }
+}
+
+/// @title BeaconRewardsEscrowBeneficiary
+/// @notice Transfer the received tokens to a designated
+///         BeaconRewardsEscrowBeneficiary contract.
+contract BeaconRewardsEscrowBeneficiary is StakerRewardsBeneficiary {
+    constructor(IERC20 _token, IStakerRewards _stakerRewards)
+        public
+        StakerRewardsBeneficiary(_token, _stakerRewards)
+    {
+        token = _token;
+        stakerRewards = _stakerRewards;
+    }
+}
+
+/// @title ECDSABackportRewardsEscrowBeneficiary
+/// @notice Trasfer the received tokens to a designated
+///         ECDSABackportRewardsEscrowBeneficiary contract.
+contract ECDSABackportRewardsEscrowBeneficiary is StakerRewardsBeneficiary {
```

Comment:
Wouldn't defining `ECDSABackportRewardsEscrowBeneficiary` and `ECDSARewardsEscrowBeneficiary` next to rewards contracts in `keep-ecdsa` be more accurate?

Reply:
One the one hand yes, but my take was this is a one time transfer of rewards and we wanted to have all beneficiaries in one place for simplicity.

Reply:
I think it's fine to have it here - `keep-core` defines all the foundation for token and how it is dispatched.

Reply:
From migrations scripts perspective it would be better to define them in `keep-ecdsa` as ECDSA beneficaires need references to ECDSA Rewards contracts.